stl/inc/flat_map

```diff
@@ -0,0 +1,300 @@
+#pragma once
+
+#include <algorithm>
+#include <initializer_list>
+#include <compare>
+#include <concepts>
+#include <cstdint>
+#include <memory>
+#include <ranges>
+#include <type_traits>
+#include <utility>
+#include <vector>
+#include <xmemory>

```

Comment:
I don't believe we should include `<cstdint>` or `<memory>`. In the future, it may be better to avoid including `<algorithm>` and `<ranges>`.

Reply:
But how do I use functions like `sort` if I don't include them?